packages/stream_chat/lib/src/api/retry_queue.dart

```diff
@@ -167,14 +167,14 @@ class RetryQueue {
     _subscriptions.forEach((s) => s.cancel());
   }
 
-  static int _byDate(Message m1, Message m2) {
-    final date1 = _getMessageDate(m1);
-    final date2 = _getMessageDate(m2);
+  static int _byDate(Message? m1, Message? m2) {
+    final date1 = _getMessageDate(m1!)!;
+    final date2 = _getMessageDate(m2!)!;
 
     return date1.compareTo(date2);
   }
 
-  static DateTime _getMessageDate(Message m1) {
+  static DateTime? _getMessageDate(Message m1) {
```

Comment:
if we can never have a null status, then it's better to not return null from this function and just suppress the lint.